command/agent/http_test.go

```diff
@@ -1056,7 +1064,16 @@ func TestHTTPServer_Limits_OK(t *testing.T) {
 			defer s.Shutdown()
 
 			assertTimeout(t, s, tc.assertTimeout, tc.timeout)
+
 			if tc.assertLimit {
+				// There's a race between assertTimeout(false) closing
+				// its connection and the HTTP server noticing and
+				// untracking it. Since there's no way to coordiante
+				// when this occurs, sleeping is the only way to avoid
+				// asserting limits before the timed out connection is
+				// untracked.
+				time.Sleep(1 * time.Second)
```

Comment:
If you can think of another way to wait until the HTTP server notices a connection has been closed, I'm all ears!

Since these tests are parallelized it doesn't seem to have an impact on overall test time.